File: packages_mjs/fetch-auth-config/src/types/auth-config.ts
import { AuthType } from './auth-type.js';
import { TokenResolverType } from './token-resolver.js';

export interface AuthResolutionMeta {
    /** Which env vars were used */
    resolvedFrom: Record<string, string>;
    /** Token resolver type used */
    tokenResolver: TokenResolverType;
    /** Whether placeholder values were used */
    isPlaceholder: boolean;
}

export interface AuthConfig {
    /** Authentication type */
    type: AuthType;

    /** Provider name from YAML */
    providerName: string;

    /** Resolution metadata */
    resolution: AuthResolutionMeta;

    /** Token/API key (for bearer, x-api-key) */
    token?: string;

    /** Username (for basic auth) */
    username?: string;

    /** Password (for basic auth) */
    password?: string;

    /** Email (for basic_email_token) */
    email?: string;

    /** Custom header name (for custom auth) */
    headerName?: string;

    /** Custom header value */
    headerValue?: string;

    // EdgeGrid Specifics
    clientToken?: string;
    clientSecret?: string;
    accessToken?: string;
    baseUrl?: string;
    headersToSign?: string[];
}

File: packages_py/fetch_auth_config/src/fetch_auth_config/types/auth_config.py
from typing import Dict, Optional, Any
from dataclasses import dataclass, field
from .auth_type import AuthType
from .token_resolver import TokenResolverType

@dataclass
class AuthResolutionMeta:
    """Metadata about how authentication configuration was resolved."""
    resolved_from: Dict[str, str] = field(default_factory=dict)
    token_resolver: TokenResolverType = TokenResolverType.STATIC
    is_placeholder: bool = False

@dataclass
class AuthConfig:
    """Structure containing resolved authentication credentials."""
    type: AuthType
    provider_name: str
    resolution: AuthResolutionMeta
    
    # Credentials
    username: Optional[str] = None
    password: Optional[str] = None
    email: Optional[str] = None
    token: Optional[str] = None
    header_name: Optional[str] = None
    header_value: Optional[str] = None
    
    # EdgeGrid Specifics
    client_token: Optional[str] = None
    client_secret: Optional[str] = None
    access_token: Optional[str] = None
    base_url: Optional[str] = None
    headers_to_sign: Optional[list] = None
